tests/predicates_declared/predicates_declared.exp

```diff
@@ -1,3 +1,76 @@
+Error ------------------------------------------------------------------------------------------ class-predicate.js:23:5
+
+Cannot extend `Hello` [1] with `D2` because predicate encoded in return [2] is incompatible with predicate encoded in
+return [3] in property `test`. [incompatible-extend]
+
+   class-predicate.js:23:5
+   23|     return typeof x === "number";
+           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [2]
+
+References:
+   class-predicate.js:21:18
+   21| class D2 extends Hello {
+                        ^^^^^ [1]
+   class-predicate.js:5:5
+    5|     return typeof x === "string";
+           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [3]
+
+
+Error ------------------------------------------------------------------------------------------ class-predicate.js:28:3
+
+Cannot extend `Hello` [1] with `D3` because function [2]. is incompatible with function [3]. in property `test`.
+[incompatible-extend]
+
+   class-predicate.js:28:3
+         v------------------------------------------------------
+   28|   test(x: mixed): boolean { // error non-predicate method
+   29|     return typeof x === "string";
+   30|   }
+         ^ [2]
+
+References:
+   class-predicate.js:27:18
+   27| class D3 extends Hello {
+                        ^^^^^ [1]
+   class-predicate.js:4:3
+         v--------------------------------
+    4|   test(x: mixed): boolean %checks {
+    5|     return typeof x === "string";
+    6|   }
+         ^ [3]
```

Comment:
The new error message is still a bit odd (all of those periods)

Reply:
The error is created in
https://github.com/facebook/flow/blob/7c7e60697a099209e4aff191b2ee8c68739104a0/src/typing/flow_js.ml#L4323-L4326
and is formatted in
https://github.com/facebook/flow/blob/7c7e60697a099209e4aff191b2ee8c68739104a0/src/typing/errors/error_message.ml#L2142-L2143